### PR TITLE
Nc news 5

### PR DESCRIPTION
**Pull request by cCody34**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -78,4 +110,15 @@ describe("app", () => {
       });
     });
   });
+  describe("ALL /notapath", () => {
```
Comment (johnhutcheon) [inline __tests__/app.test.js:113]: Nicely done, well done on putting this in it's own describe block, as it's not specific to just one ticket 👍

File: db/controllers/articles-controller.js [hunk 2]
```diff
@@ -1,12 +1,26 @@
...
+const { readComments } = require("../models/comments-models");
+
+const getArticles = (req, res, next) => {
+  const promises = [readArticles(), readComments()];
+  return Promise.all(promises).then(([articles, comments]) => {
+    articles.forEach((article) => {
+      const articleComments = comments.filter((comment) => {
+        return comment.article_id === article.article_id;
+      });
+      article.comment_count = articleComments.length
```
Comment (johnhutcheon) [inline db/controllers/articles-controller.js]: This is cleverly done, however, we shouldn't be doing this sort of thing in the controller. We can make a single query in the articles model that gets all of the articles, then joins the table with the comments one, to create the comment_count. 
We can then just invoke the articles model here, which will give us all the required information. 👍

File: db/models/articles-model.js [hunk 3]
```diff
@@ -1,5 +1,14 @@
 const db = require("../connection");
 
+const readArticles = () => {
+  return db.query(`SELECT * FROM articles ORDER BY created_at DESC;`).then(({ rows }) => {
```
Comment (johnhutcheon) [inline db/models/articles-model.js]: As per my previous comment, please amend this SQL query so it joins the comments table and creates the comment_count. 🙂

File: db/models/articles-model.js [hunk 4]
```diff
@@ -1,5 +1,14 @@
...
+    rows.forEach((article) => {
+      delete article.body;
```
Comment (johnhutcheon) [inline db/models/articles-model.js]: The SQL query can allow us to select which columns we want, so you shouldn't have to delete article.body here when you refactor it. 👍

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Hey Charlotte, good work, but I'm going to ask you to resubmit please. If you see my comments, rather than having all the logic for creating the comment_count in your controller, this can be done in your model via a single SQL query. Take a look at the "Group by and aggregate functions" examples in the SQL relationships notes. If you get stuck though, give us a shout. 👍

**Comment by Harrison365** [comment]
Great work Charlotte 🎉 
Merge away ✅

**Pull request merged**